### DISCUSSION TITLE
Only small number of aligned reads are used to assemble in “generic ht single cell fragmented”

### DISCUSSION THREAD

**Discussion by bigcat1001** [milaboratory/mixcr#1590] [Q&A] [answered]
Hi!
I have a strange question when using preset “generic ht single cell fragmented”.
The alignment looks fine, aligned reads: 2266062 (66.37%),but only 6.36% aligned reads aligned to CDR3.
And I can see in the assembly report that the number of reads used in the final clone is not very high, reads used in clones, percentage of total: 296775 (8.69%),which indicates that a large number of reads were not used in the assembly
How can I see where other reads have been aligned?

> Analysis time: 1.71m
Total sequencing reads: 3414365
Successfully aligned reads: 2266062 (66.37%)
Coverage (percent of successfully aligned):
  CDR3: 144201 (6.36%)
  FR3_TO_FR4: 10312 (0.46%)
  CDR2_TO_FR4: 8026 (0.35%)
  FR2_TO_FR4: 175 (0.01%)
  CDR1_TO_FR4: 39 (0%)
  VDJRegion: 92 (0%)
Alignment failed: no hits (not TCR/IG?): 1134694 (33.23%)
Alignment failed after alignment-aided overlap: 6 (0%)
Alignment failed: low total score: 13603 (0.4%)
Overlapped: 2026366 (59.35%)
Overlapped and aligned: 1743232 (51.06%)
Overlapped and not aligned: 283134 (8.29%)
Alignment-aided overlaps, percent of overlapped and aligned: 44056 (2.53%)
No CDR3 parts alignments, percent of successfully aligned: 1912069 (84.38%)
Partial aligned reads, percent of successfully aligned: 209792 (9.26%)
V gene chimeras: 272732 (7.99%)
J gene chimeras: 38 (0%)
Paired-end alignment conflicts eliminated: 1713 (0.05%)
Realigned with forced non-floating bound: 2864110 (83.88%)
Realigned with forced non-floating right bound in left read: 38953 (1.14%)
Realigned with forced non-floating left bound in right read: 38953 (1.14%)
IGH chains: 2266062 (100%)
IGH non-functional: 1481 (0.07%)
Trimming report:
  R1 reads trimmed left: 24 (0%)
  R1 reads trimmed right: 147 (0%)
  Average R1 nucleotides trimmed left: 3.7722973378651666E-4
  Average R1 nucleotides trimmed right: 5.97182785085953E-4
  R2 reads trimmed left: 636 (0.02%)
  R2 reads trimmed right: 476 (0.01%)
  Average R2 nucleotides trimmed left: 7.644173953282675E-4
  Average R2 nucleotides trimmed right: 3.2041096953606306E-4
Tag parsing report:
  Execution time: 0ns
  Total reads: 3414365
  Matched reads: 3414365 (100%)
  Projection +R1 +R2: 3414365 (100%)
  For variant 0:
    For projection +R1 +R2:
      R1:Left position: 7
      R1:Right position: 150
      CELL:Left position: 0
      R2:Left position: 22
      CELL:Right position: 22
      Variants: 0
      Cost: 0
      R1 length: 143
      CELL length: 22
      R2 length: 128
======================================

> Analysis time: 59.76s
Final clonotype count: 17661
Reads used in clonotypes, percent of total: 296775 (8.69%)
Average number of reads per clonotype: 16.8
Reads dropped due to the lack of a clone sequence, percent of total: 1729586 (50.66%)
Reads dropped due to a too short clonal sequence, percent of total: 6 (0%)
Reads dropped due to low quality, percent of total: 0 (0%)
Reads dropped due to failed mapping, percent of total: 73604 (2.16%)
Reads dropped with low quality clones, percent of total: 1797 (0.05%)
Aligned reads processed: 372596
Reads used in clonotypes before clustering, percent of total: 297189 (8.7%)
Number of reads used as a core, percent of used: 297189 (100%)
Mapped low quality reads, percent of used: 0 (0%)
Reads clustered in PCR error correction, percent of used: 0 (0%)
Reads pre-clustered due to the similar VJC-lists, percent of used: 0 (0%)
Clonotypes dropped as low quality: 219
Clonotypes eliminated by PCR error correction: 0
Clonotypes pre-clustered due to the similar VJC-lists: 0
Clones dropped in post filtering: 130 (0.73%)
Reads dropped in post filtering: 414.0 (0.01%)
Alignments filtered by tag prefix: 272 (0.01%)
IGH chains: 17661 (100%)
IGH non-functional: 173 (0.98%)
Pre-clone assembler report:
  Number of input groups: 160412
  Number of input groups with no assembling feature: 118784
  Number of input alignments: 1853202
  Number of alignments with assembling feature: 123616 (6.67%)
  Number of output pre-clones: 51286
  Number of pre-clonotypes per group:  
    0: + 26 (0.06%) = 26 (0.06%)
    1: + 34480 (82.83%) = 34506 (82.89%)
    2: + 5474 (13.15%) = 39980 (96.04%)
    3: + 1118 (2.69%) = 41098 (98.73%)
    4: + 281 (0.68%) = 41379 (99.4%)
    5: + 114 (0.27%) = 41493 (99.68%)
    6: + 135 (0.32%) = 41628 (100%)
  Number of assembling feature sequences in groups with zero pre-clonotypes: 3641
  Number of dropped pre-clones by tag suffix conflict: 0
  Number of dropped alignments by tag suffix conflict: 0
  Number of core alignments: 119092 (6.43%)
  Discarded core alignments: 4524 (3.8%)
  Empirically assigned alignments: 253253 (13.67%)
  Empirical assignment conflicts: 1584 (0.09%)
  Tag+VJ-gene empirically assigned alignments: 254837 (13.75%)
  VJ-gene empirically assigned alignments: 0 (0%)
  Tag empirically assigned alignments: 0 (0%)
  Number of ambiguous groups: 7122
  Number of ambiguous V-genes: 6431
  Number of ambiguous J-genes: 4830
  Number of ambiguous tag+V/J-gene combinations: 11261
  Ignored non-productive alignments: 0 (0%)
  Unassigned alignments: 320202 (17.28%)
======================================

**Comment by mizraelson** [comment, marked as answer]
Hi,
Since the data is fragmented (judging by the preset you have used) it is expected that only a small part of reads cover CDR3 region.  Only 6.36% percent of 66% of total reads covers CDR3. The rest of the 66% cover other regions.
For this preset clones are initially assembled by CDR3 sequence, that is why assemble report shows a low number of reads used. Then `assembleContigs` (the next step in the pipeline) extends the initial CDR3 clones to cover the rest of the receptor sequence using the reads that were not used during the assembly.

**Comment by bigcat1001** [comment]
Thanks for you  reply
AssembleContigs.report is below.But i didn't find how many reads was used.How can I determined the assemble quality? 

> Analysis time: 19.95s
Initial clonotype count: 17661
Final clonotype count: 17661 (100%)
Canceled assemblies: 0 (0%)
Number of premature termination assembly events, percent of number of initial clonotypes: 68.0 (0.39%)
Longest contig length: 752
Clustered variants: 0 (0%)
Reads in clustered variants: 0.0 (0%)
Reads in divided (newly created) clones: 0.0 (0%)
Clones with ambiguous letters in splitting region: 0 (0%)
Reads in clones with ambiguous letters in splitting region: 0.0 (0%)
Average number of ambiguous letters per clone with ambiguous letters in splitting region: NaN

**Reply by mizraelson** [reply]
what number do you get if you sum up the readCounts in the clonotype table?

**Comment by bigcat1001** [comment]
Thanks,I ignored this.Only 296775 reads were used for final clones which is about 10% of alignment reads(296775/2266062 ).
I will try to extract some reads to check what happened.
Based on 
> Coverage (percent of successfully aligned):
CDR3: 144201 (6.36%)
FR3_TO_FR4: 10312 (0.46%)
CDR2_TO_FR4: 8026 (0.35%)
FR2_TO_FR4: 175 (0.01%)
CDR1_TO_FR4: 39 (0%)
VDJRegion: 92 (0%)

I suppose that too many reads just cover 5' UTR? Do you have any suggestion?

**Reply by mizraelson** [reply]
This part doesn't say anything about 5'UTR. It shows that 6% cover CDR3, and then the longer the region the lower the percentage of reads cover it which is expected with fragmented data. What protocol did you use?

**Comment by bigcat1001** [comment]
The protocol I used similar to SMARTer3. Briefly,I prepared full length library using 5' race(similar to the TAKARA kit).Then,Tn5 with adaptors was added to fragment the full length  library and a low  number of third PCR cycles were used to amplify the fragment and sequcing using PE150.The final library structrue is just like single cell.

**Reply by bigcat1001** [reply]
The aim of this protocol is try to recover full length VDJ based on PE150 platform,not PE300.

**Reply by mizraelson** [reply]
But is it a single cell library? Or the barcode is a UMI?

**Reply by bigcat1001** [reply]
No,it is a bulk library.The barcode is a UMI. When i try the preset.I set UMIs as cell barcode.

**Reply by mizraelson** [reply]
Can you try the following: unzip the yaml file attached, put it into the directory you run MiXCR from (or to `~/.mixcr/presets/`) and run the command:

```
mixcr analyze local:generic-shotgun-with-umi \
--species hsa \
--tag-pattern "^N{7}(R1:*)\^(UMI:N{22}(R2:*)"
input_R1.fastq.gz \
input_R2.fastq.gz \
result
```

Adjust species and pattern if needed (but use UMI not CELL barcode).

[generic-shotgun-with-umi.yaml.zip](https://github.com/milaboratory/mixcr/files/14611999/generic-shotgun-with-umi.yaml.zip)
